### ISSUE TITLE
Add a way to deal with exceptions in rules

### ISSUE THREAD

**Issue by micheleorselli** [phparkitect/arkitect#114]
e.g a domain class must depend only on other domain classes, but also it can depend on UuidInterface

**Comment by micheleorselli** [comment]
@fain182 @AlessandroMinoccheri  we could add an "ignore" method to our DSL, which will intervene when evaluating a rule, deciding if a rule for a given class should be skipped. What do you think? The resulting api would be something like

```php
ClassSet::allClasses()
->that(new ResideInNamespace('*\Controller\*'))
->should(new DependsOnlyOnClassesInNamespaces('*\Controller\*'))
->ignore('UuIdInterface', 'MyVendor\Myclass', ...)
->because('we want clean controllers')

```

I find this pretty readable, even if it is a bit ambiguous, since we don't know if ignores applies to ->that ->should, or both. We could overcome this problema following another approach, adding a second parameter to should. 
The result would be something like

```php
ClassSet::allClasses()
->that(new ResideInNamespace('*\Controller\*'))
->should(new DependsOnlyOnClassesInNamespaces('*\Controller\*'), 'UuIdInterface', 'MyVendor\Myclass')
->because('we want clean controllers')

```

This solves the ambiguity problem, though I find it harder to read

**Comment by AlessandroMinoccheri** [comment]
I think that the first option is more readable and clear for someone that would like to use this feature.

**Comment by fain182** [comment]
I try to take the best part of both:

```
ClassSet::allClasses()
->that(new ResideInNamespace('*\Controller\*'))
->should(
    new DependsOnlyOnClassesInNamespaces(
        '*\Controller\*', 
        ['except' => [UuIdInterface', 'MyVendor\Myclass']]
    ))
->because('we want clean controllers')
```

For me it's a concept of the rule, not of the DSL...

**Comment by micheleorselli** [comment]
let's try to change the expression DependsOnlyOnClassesInNamespaces to accept both namespaces and classes. eg

```php
new DependsOnlyOnClassesInNamespaces('*\Controller\*', 'UuidInterface')
```

**Comment by AlessandroMinoccheri** [comment]
Solved by #219
